cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs

```diff
@@ -187,7 +187,7 @@ instance Show (Hash h a) where
 
 instance HashAlgorithm h => IsString (Hash h a) where
   fromString str =
-    case hashFromBytesAsHex (BSC.pack str) of
+    case hashFromBytesAsHex (BSC.pack $ read str) of
```

Comment:
I suspect this is not what we want:

```
λ> "a4612527d3b6018c5290118c48e62fc1" :: Hash MD5 [Char]
"*** Exception: Prelude.read: no parse
λ> "\"a4612527d3b6018c5290118c48e62fc1\"" :: Hash MD5 [Char]
"a4612527d3b6018c5290118c48e62fc1"
it :: Hash MD5 [Char]
(0.00 secs, 1,053,712 bytes)
```

Reply:
Right; it's better to fix the `Show` instance.